src/scripts.c

```diff
@@ -49,37 +50,60 @@ static void update_boot(void)
 
 static void update_triggers(bool block)
 {
-	char *restart_cmd = NULL;
-	char *block_flag = NULL;
+	char *cmd = NULL;
+	char const *block_flag = NULL;
+	char const *reexec_flag = NULL;
 	__attribute__((unused)) int ret = 0;
 
-	/* Check that systemd is working (container case)
-	 * and return if it is not after printing an error */
-	ret = system("/usr/bin/systemctl > /dev/null 2>&1");
-	if (ret != 0) {
-		fprintf(stderr, "WARNING: systemctl not operable, "
+	if (strlen(POST_UPDATE) == 0) {
+		/* fall back to systemd if path prefix is not the rootfs
+		 * and the POST_UPDATE trigger wasn't specified */
+		if (strcmp("/", path_prefix) != 0) {
+			return;
+		}
+
+		ret = system("/usr/bin/systemctl > /dev/null 2>&1");
+		if (ret != 0) {
+			fprintf(stderr, "WARNING: systemctl not operable, "
 				"unable to run systemd update triggers\n");
-		return;
-	}
+			return;
+		}
+		/* These must block so that new update triggers are executed after */
+		if (need_systemd_reexec) {
+			ret = system("/usr/bin/systemctl daemon-reexec");
+		} else {
+			ret = system("/usr/bin/systemctl daemon-reload");
+		}
+		if (!block) {
+			block_flag = "--no-block";
+		} else {
+			block_flag = "";
+		}
 
-	/* These must block so that new update triggers are executed after */
-	if (need_systemd_reexec) {
-		ret = system("/usr/bin/systemctl daemon-reexec");
+		string_or_die(&cmd, "/usr/bin/systemctl %s restart update-triggers.target", block_flag);
 	} else {
-		ret = system("/usr/bin/systemctl daemon-reload");
-	}
+		/* These must block so that new update triggers are executed after */
+		if (need_systemd_reexec) {
+			reexec_flag = "--reexec";
+		} else {
+			reexec_flag = "";
+		}
 
-	if (!block) {
-		string_or_die(&block_flag, "--no-block");
-	} else {
-		string_or_die(&block_flag, "");
-	}
+		if (!block) {
+			block_flag = "--no-block";
+		} else {
+			block_flag = "";
+		}
```

Comment:
Since you do this in both the `if` and `else` clauses, how about setting the `block_flag` in one place, before the conditional?

Reply:
Sure. And fix those tabs...